pkg/agent/globalstate.go

```diff
@@ -172,9 +146,170 @@ func (s *agentState) handleVMEventAdded(
 		status:        status,
 		vmInfoUpdated: txVMUpdate,
 	}
+	s.metrics.runnerStarts.Inc()
 	runner.Spawn(runnerCtx, rxVMUpdate)
 }
 
+// FIXME: make these timings configurable.
+const (
+	RunnerRestartMinWaitSeconds = 5
+	RunnerRestartMaxWaitSeconds = 10
+)
+
+func (s *agentState) TriggerRestartIfNecessary(runnerCtx context.Context, podName util.NamespacedName, podIP string) {
```

Comment:
I think `context.Context` objects should be named `ctx`, particularly the ones that get passed to a function unless there's a good reason for it, which might exist but it's not obvious (and I read the function looking for it because I was suspicious.)

Reply:
Added a comment above the function:

> NB: runnerCtx is the context *passed to the new Runner*. It is only used here to end our restart process early if it's already been canceled.

lmk if that's suitable